observability/src/main/java/io/grpc/observability/logging/GcpLogSink.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.observability.logging;
+
+import com.google.cloud.MonitoredResource;
+import com.google.cloud.logging.LogEntry;
+import com.google.cloud.logging.Logging;
+import com.google.cloud.logging.LoggingOptions;
+import com.google.cloud.logging.Payload.JsonPayload;
+import com.google.cloud.logging.Severity;
+import com.google.common.base.Strings;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+import io.grpc.internal.JsonParser;
+import io.grpc.observabilitylog.v1.GrpcLogRecord;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Sink for Google Cloud Logging.
+ */
+public class GcpLogSink implements Sink {
+
+  private static final Logger logger = Logger.getLogger(Sink.class.getName());
+
+  private static final String SERVICE_TO_EXCLUDE = "google.logging.v2.LoggingServiceV2";
+  private static final String DEFAULT_LOG_NAME = "grpc";
+
+  private static GcpLogSink gcpLogSinkInstance = null;
+  private final LoggingOptions gcpLoggingOptions;
+  private final Logging gcpLoggingClient;
+
+  /**
+   * Constructor for custom sink.
+   *
+   * @param destinationProjectId cloud project id to write logs
+   */
+  private GcpLogSink(String destinationProjectId) {
+    if (Strings.isNullOrEmpty(destinationProjectId)) {
+      gcpLoggingOptions = LoggingOptions.getDefaultInstance();
+    } else {
+      gcpLoggingOptions = LoggingOptions.newBuilder().setProjectId(destinationProjectId).build();
+    }
+    gcpLoggingClient = gcpLoggingOptions.getService();
+  }
+
+  /**
+   * Initialize GcpLogSInk.
+   *
+   * @param projectId cloud project id to write logs
+   */
+  // TODO(dnvindhya) read the projectId value from config instead of taking
```

Comment:
The TODO comment should be where the call to `init` is. I suspect `init` will still take this argument but the caller in `Observability` will read the config and pass the appropriate value instead of passing `null` as it does today.

Reply:
Removed TODO comment as we still take `projectId` as an argument.